compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -261,6 +266,221 @@ static CodeGenPipeline getTensorCorePipeline(Type elementType) {
   return codegenPipeline;
 }
 
+//====---------------------------------------------------------------------===//
+// Vector Distribution Reduction Pipeline Configuration
+//====---------------------------------------------------------------------===//
+
+/// Check if `op` is a linalg.reduce or a linalg.generic that has at least one
+/// reduction iterator.
+static bool hasReductionIterator(linalg::LinalgOp &op) {
+  return isa<linalg::ReduceOp, linalg::GenericOp>(op) &&
+         llvm::any_of(op.getIteratorTypesArray(), linalg::isReductionIterator);
+}
+
+// Get the bitwidth of the operation.
+static FailureOr<int64_t> getBitWidth(linalg::LinalgOp op) {
+
+  Value init = op.getDpsInitOperand(0)->get();
+  Value src = op.getDpsInputOperand(0)->get();
+  Type initElemType = getElementTypeOrSelf(init);
+  Type srcElemType = getElementTypeOrSelf(src);
+
+  if (auto initOp = init.getDefiningOp<linalg::GenericOp>()) {
+    if (IREE::LinalgExt::isBitExtendOp(initOp))
+      initElemType = getElementTypeOrSelf(initOp.getDpsInputs()[0]);
```

Comment:
IREE prefers braces around the bodies of all statements (unlike llvm).

```suggestion
    if (IREE::LinalgExt::isBitExtendOp(initOp)) {
      initElemType = getElementTypeOrSelf(initOp.getDpsInputs()[0]);
    }
```

Also elsewhere below